tests/suites/test_suite_psa_crypto.function

```diff
@@ -948,8 +949,25 @@ void import_export( data_t *data,
     psa_set_key_algorithm( &attributes, alg );
     psa_set_key_type( &attributes, type );
 
+    if( PSA_KEY_TYPE_IS_DH( type ) &&
+        expected_export_status == PSA_ERROR_BUFFER_TOO_SMALL )
+    {
+        export_size -= 8;
+    }
+
+    if( PSA_KEY_TYPE_IS_DH( type ) &&
+          ( data->len != 256 && data->len != 384 &&
+            data->len != 512 && data->len != 768 && data->len != 1024 ) )
+    {
+        expected_import_result =  PSA_ERROR_INVALID_ARGUMENT;
+    }
+
     /* Import the key */
-    PSA_ASSERT( psa_import_key( &attributes, data->x, data->len, &key ) );
+    TEST_EQUAL( psa_import_key( &attributes, data->x, data->len, &key ),
+                expected_import_result );
+
+    if( expected_import_result != PSA_SUCCESS)
```

Comment:
if statement before test_equal? Or if statement is after test_equal for different reason?

Reply:
We continue test execution only if import was successful. If export has failed we check if failed error is as expected and jump to exit.

Reply:
Right. Thanks for explaining!